alternator/executor.cc

```diff
@@ -1897,10 +1979,18 @@ future<executor::request_return_type> executor::delete_item(client_state& client
     auto op = make_shared<delete_item_operation>(_proxy, std::move(request));
     tracing::add_table_name(trace_state, op->schema()->ks_name(), op->schema()->cf_name());
     const bool needs_read_before_write = op->needs_read_before_write();
+
+    if (!co_await client_state.check_has_permission(auth::command_desc(
```

Comment:
I thought about this - indeed this series adds more-or-less the same 3 lines of code "duplicated" in a dozen places. BUT, there are several problems:
1. Not much is duplicated, it's basically just a call to one function (grant_applicable_permissions) plus a second function to build its parameter - but that second function is actually different in different places (some places need a specific table, other use ALL KEYSPACES, for example).
2. In every case, if the if() succeeds, I do a "return" (which is slightly more efficient than throwing), so I can't even put this part in the helper function and it has to be in the caller anyway.

All in all, I decided that a helper function wen't help much.

If I'll switch to _throwing_ instead of returning, then I agree - a helper function will be nicer. Maybe I should decide that it's fine to throw on a AccessDeniedException, which should be rare?

Reply:
Introducing a wrapper makes more sense if we add additional checks, e.g. a check if authz is enabled, per https://github.com/scylladb/scylladb/pull/19762#discussion_r1686260522.

> Maybe I should decide that it's fine to throw on a AccessDeniedException, which should be rare?

What's the difference between returning an error and throwing an exception? In either case, I agree that insufficient permissions event should be rare, but only after a series of trial and error where someone tries to get the code working.

Reply:
> What's the difference between returning an error and throwing an exception?

There's soe performance difference. In general it's bad to throw exceptions if we can avoid it, even for error messages. But the truth is that there's a big risk in throwing exceptions to indicate timeout errors (@avikivity keeps warning us about the cascade of exceptions that can result in that case) but probably (?) no risk on throwing exceptions on malformed requests.

Reply:
I'm not adding a helper for now. I'm not sure it will be better (avoid duplicating a couple of lines a dozen times) or worse (can't use return, need to start using exception) so leaving the decision for later.

Reply:
Since @kostja asked me to modify the error messages, it's a good time to write this helper. I decided that using an exception is actually good, because the performance we really care about here is the performance of the **success** case (when permissions are granted) - and this performance is made better by using exceptions (at the downside of making unsuccessful operations a bit slower).